### ISSUE TITLE
Can't add a pull image secret to ActiveMQArtemis initImage

### ISSUE THREAD

**Issue by paolabassi** [artemiscloud/activemq-artemis-operator#162]
When I public the initImage on a private registry, I can't add also a pull image secret so I'm not authorized to pull the image.

**To Reproduce**
1. Add a custom initImage to a private registry
2. Define a yaml file to configure an ActiveMQArtemis broker and set the initImage url to define custom configurations
3. Try to pull the image from the private registry. I'm not authorized (no image pull secret is defined)

**Platform:**
 - OS: Linux
 - Version Ubuntu 20.04
 - Go 1.17.6

**Expected behavior**
I would like to define the initImage and also a pull image secret in the ActiveMQArtemis definition

**Comment by brusdev** [comment]
@paolabassi this issue makes sense to me, are you planning to contribute with a pull request?

**Comment by paolabassi** [comment]
> @paolabassi this issue makes sense to me, are you planning to contribute with a pull request?

No, I’m sorry. I abandoned the use of the operator because it currently has too many limitations

**Comment by naude-r** [comment]
@brusdev we've been hit by the same issue. if you can provide a rough guideline on where to add what i can take a stab at providing a pull request.

**Comment by brusdev** [comment]
@naude-r I would add the `ImagePullSecrets` field to the [DeploymentPlanType](https://github.com/artemiscloud/activemq-artemis-operator/blob/v1.0.11/api/v1beta1/activemqartemis_types.go#L288) struct that the [ActiveMQArtemisReconciler](https://github.com/artemiscloud/activemq-artemis-operator/blob/v1.0.11/controllers/activemqartemis_reconciler.go) could use to fill the field `ImagePullSecrets` of [podSpec](https://github.com/artemiscloud/activemq-artemis-operator/blob/v1.0.11/controllers/activemqartemis_reconciler.go#L1513).

**Comment by naude-r** [comment]
@brusdev thanks. will take a look.

**Comment by brusdev** [comment]
Are you using go 1.17?

**Comment by naude-r** [comment]
yes. 
```
activemq-artemis-operator]$ go1.17 version
go version go1.17 linux/amd64
```

Makefile was updated to use go1.17

**Comment by naude-r** [comment]
> yes.
> 
> ```
> activemq-artemis-operator]$ go1.17 version
> go version go1.17 linux/amd64
> ```
> 
> Makefile was updated to use go1.17

got it working. had to remove all previous go packages from a newer version.

**Comment by naude-r** [comment]
PR #582 added.

**Comment by brusdev** [comment]
@naude-r thanks, I have added some comments to your PR #582

**Comment by brusdev** [comment]
This enhancement will be included in v1.0.12
